### PR TITLE
Elaboration

### PR DESCRIPTION
**Pull request by krushkaa**
Added entities, resources and pom file
Than controllers, repositories, services, security added

### CODE REVIEW

**Comment by krushkaa** [comment]
Attempting to implement the registration method, but the “registration” page does not open on the /welcomePage start page.

File: pom.xml [hunk 1]
```diff
@@ -89,6 +80,16 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (DHrytsuk) [inline pom.xml:85]: What is this?

File: src/main/java/org/example/parampammusic/entity/Role.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int role_id;
+
+    @Column(name = "role_name", nullable = false, unique = true)
+    private String roleName;
+
+    @OneToMany(mappedBy = "role", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<User> user = new ArrayList<>();
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/entity/Role.java:23]: This field is not needed here

File: src/main/java/org/example/parampammusic/entity/User.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                ", login='" + login + '\'' +
+                ", password='" + password + '\'' +
+                ", email='" + email + '\'' +
+                ", telNumber='" + telNumber + '\'' +
+                ", bonusPoints=" + bonusPoint +
+                ", role=" + role +
+                '}';
+    }
+
+    public User(int user_id, String username, String password, String email, String telNumber, int bonusPoints, Role role) {
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/entity/User.java:63]: WHy do you need it?

File: src/main/java/org/example/parampammusic/entity/OrderItem.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    private Order order;
+
+    @ManyToOne
+    @JoinColumn(name = "audio_track_id", nullable = false)
+    private AudioTrack audioTrack;
+
+    @Column(name = "price", nullable = false)
+    private double price;
+
+    public OrderItem(Order order, AudioTrack track) {
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/entity/OrderItem.java:31]: What is this?

File: src/main/java/org/example/parampammusic/entity/OrderItem.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+import jakarta.persistence.*;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@EqualsAndHashCode
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/entity/OrderItem.java:11]: Please, use same approach for all classes: you should either use annotations or override equals and hashCode.
For constructors the same note!

File: src/main/java/org/example/parampammusic/entity/Genre.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@EqualsAndHashCode
+@ToString
+@Entity
+@Table(name = "genre")
+public class Genre {
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/entity/Genre.java:15]: Where is this entity is used?

File: src/main/java/org/example/parampammusic/config/ThymeleafConfig.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+public class ThymeleafConfig implements WebMvcConfigurer {
+
+    @Override
+    public void addViewControllers(ViewControllerRegistry registry) {
+        registry.addViewController("/loginForm").setViewName("loginForm");
+        registry.addViewController("/registrationForm").setViewName("registration");
+    }
+
+    @Bean
+    public SpringResourceTemplateResolver templateResolver() {
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/config/ThymeleafConfig.java:22]: Do you need it for your viewvs?
You are using spring boot and can do in application file

File: src/main/java/org/example/parampammusic/controller/Model.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+import lombok.Data;
+import lombok.experimental.Accessors;
+import org.example.parampammusic.entity.OrderItem;
+import org.example.parampammusic.entity.User;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class Model {
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/controller/Model.java:11]: Explain, what is this class for?

File: src/main/java/org/example/parampammusic/repository/RoleRepository.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package org.example.parampammusic.repository;
+
+import org.example.parampammusic.entity.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface RoleRepository extends JpaRepository<Role, Long> {
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/repository/RoleRepository.java:10]: It seems, in Role entity your primary key is Integer. WHy do you have Long here?

File: src/main/java/org/example/parampammusic/service/UserServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    @Autowired
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/service/UserServiceImpl.java:18]: Please, use same approach for all classes

File: src/main/java/org/example/parampammusic/service/UserServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    private UserRepository userRepository;
+
+    @Override
+    public User createUser(UserDTO userDTO) {
+        User user = new User();
+        user.setLogin(userDTO.getLogin());
+        user.setPassword(userDTO.getPassword());
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/service/UserServiceImpl.java:25]: We need to store encoded password in the database

File: src/main/java/org/example/parampammusic/security/SecurityConfig.java [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+                .authorizeHttpRequests(request -> request
+                        .requestMatchers("/admin/**").hasRole("ADMIN")
+                        .requestMatchers("/user/**").hasRole("USER")
+                        .requestMatchers("/welcomePage", "/registration", "/loginForm", "/logout").permitAll()
+                        .dispatcherTypeMatchers(DispatcherType.FORWARD).permitAll()
+                )
+                .formLogin(formLogin -> formLogin
+                        .loginPage("/loginForm")
+                        .loginProcessingUrl("/login")
+                        .usernameParameter("login")
```
Comment (DHrytsuk) [inline src/main/java/org/example/parampammusic/security/SecurityConfig.java:30]: I see, that this field called `username` in login form. Why do you have `login` here?

**Comment by DHrytsuk** [comment]
@krushkaa , do you have correct mapping in security config to open registration page?

**Comment by krushkaa** [comment]
> @krushkaa , do you have correct mapping in security config to open registration page?

You and I have been fixing code in class. I'm still working on the mistakes, I'll upload the corrected code soon.

File: src/main/java/org/example/parampammusic/repository/RoleRepository.java (hunk 9 above)
Comment (DHrytsuk) [reply src/main/java/org/example/parampammusic/repository/RoleRepository.java:10]: Not fixed

**Pull request closed without merging**